### PR TITLE
chore(deps): angular 17 support

### PR DESCRIPTION
**Pull request by Supamiu**
## Description

Update Angular dependencies to 17.0.X

-----

This required a full rewrite of the `msw` handlers to use `msw` 2.0 instead so it can work together with a more recent version of Typescript.

-----

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2410/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/9507/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2410/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/9515/ui-diff)

File: package.json [hunk 1]
```diff
@@ -88,24 +89,24 @@
...
-    "msw": "^1.2.1",
-    "ng-packagr": "^16.0.0",
+    "jest-preset-angular": "^13.1.4",
+    "msw": "^2.0.6",
+    "ng-packagr": "^17.0.0",
     "postcss": "^8.4.23",
     "postcss-scss": "^4.0.6",
     "postcss-selector-parser": "^6.0.12",
     "postcss-value-parser": "^4.2.0",
     "prettier": "^2.8.8",
```
Comment (thomasmarcelin) [inline package.json:99]: `prettier` v3.1 supporte le nouveau control flow normalement, peut être que ça vaudrait le coup de faire la migration ici ? 
https://prettier.io/blog/2023/11/13/3.1.0

Comment (Supamiu) [reply package.json:99]: Vu qu'on ne l'utilise pas pour l'instant, je pense que ça vaut le coup d'attendre parce que control flow est en developer preview et que de toute façon, quand on passera au control flow de v17, prettier aura sûrement un encore meilleur support pour 🙂

**Review by thomasmarcelin — DISMISSED** [review]

File: .storybook/main.mjs [hunk 2]
```diff
@@ -1,8 +1,9 @@
+import { dirname, join } from "path";
 import { URL, fileURLToPath } from 'url';
 
 export default {
 	framework: {
-		name: '@storybook/angular',
+		name: getAbsolutePath("@storybook/angular"),
```
Comment (TanukiSharp) [inline .storybook/main.mjs:6]: Nit: Don't we care about single quote consistency for strings ?

File: package.json [hunk 3]
```diff
@@ -35,49 +35,50 @@
...
-    "@angular/common": "^16.1.3",
-    "@angular/compiler": "^16.1.3",
-    "@angular/core": "^16.1.3",
-    "@angular/forms": "^16.1.3",
-    "@angular/platform-browser": "^16.1.3",
-    "@angular/platform-browser-dynamic": "^16.1.3",
-    "@angular/router": "^16.1.3",
+    "@angular/animations": "^17.0.2",
+    "@angular/cdk": "^17.0.0",
+    "@angular/common": "^17.0.2",
```
Comment (shajz) [inline package.json:40]: We must reflect these bumps in the ng/package.json file to allow apps to update to Angular 17.
(GitHub won't let me comment on the whole file, sorry)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2410/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/9651/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2410/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/9694/ui-diff)

**Review by GuillaumeNury — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -35,50 +35,51 @@
...
+    "@storybook/addon-mdx-gfm": "^7.5.3",
+    "@storybook/angular": "^7.5.3",
+    "@storybook/test-runner": "^0.14.0",
+    "@testing-library/angular": "^14.4.0",
+    "@testing-library/jest-dom": "^6.1.4",
+    "@testing-library/user-event": "^14.5.1",
     "@types/dompurify": "^3.0.2",
     "@types/jest": "^29.5.1",
     "@types/jest-axe": "^3.5.5",
     "@types/node": "^18.11.10",
```
Comment (shajz) [inline package.json:80]: Not very important but this should be bumped to 18.17

Comment (Supamiu) [reply package.json:80]: I'll do this in another PR to not stale previous approves ^^

**Pull request merged**